pypeit/display/ginga_spec1dview.py

```diff
@@ -0,0 +1,380 @@
+# This is open-source software licensed under a BSD license.
+# Please see the file LICENSE.txt for details.
+"""
+Spec1dView is a plugin for the Ginga viewer that provides functionality for
+visualizing and analyzing 1D spectra from FITS files. The plugin allows users
+to plot spectra, identify spectral lines from various line lists, and
+customize the display according to different parameters.
+
+**Plugin Type: Local**
+
+Spec1dView is a local plugin, which means it is associated with a specific
+channel in the Ginga viewer. An instance of the plugin can be opened for
+each channel, allowing for multiple spectra to be analyzed simultaneously.
+
+**Usage**
+- Load and visualize 1D spectra from FITS files.
+- Customize the display by selecting different line lists, extraction types,
+  and flux/mask settings.
+- Update the redshift to shift the spectral lines accordingly.
+
+**Editing**
+Users can modify the visualization by:
+- Choosing from a variety of line lists to identify spectral features.
+- Selecting different types of extraction methods (OPT, BOX).
+- Applying or removing flux calibration and masking options.
+- Updating the redshift value to reflect the observed wavelengths.
+
+**UI**
+The user interface provides controls for:
+- Selecting the line list from a combobox.
+- Entering a redshift value to shift the spectrum.
+- Choosing the extraction type, flux calibration, and masking options via comboboxes.
+- Buttons to load a FITS file and clear the current selection.
+
+**Buttons**
+- Update z: Updates the redshift value and refreshes the spectrum plot.
+- Enter: Loads the specified FITS file for analysis.
+- Clear: Clears the current inputs and resets the UI settings.
+
+**Tips**
+- Use the comboboxes to switch between different line lists and adjust the spectrum display settings.
+- Ensure that the correct FITS file path is entered before attempting to load the data.
+"""
+import time
+import numpy as np
+
+from ginga import GingaPlugin
+from ginga.misc import Bunch
+from ginga.gw import Widgets
+from ginga.table.AstroTable import AstroTable
+from ginga.plot.Plotable import Plotable
+from ginga.canvas.CanvasObject import get_canvas_types
+
+from pypeit import specobjs
+from pypeit import utils
+# TODO: need to make PypeIt LineList class to deprecate linetools
+from linetools.lists.linelist import LineList
+
+__all__ = ['Spec1dView']
+
+
+class Spec1dView(GingaPlugin.LocalPlugin):
+
+    def __init__(self, fv, fitsimage):
+        # superclass defines some variables for us, like logger
+        super(Spec1dView, self).__init__(fv, fitsimage)
```

Comment:
Nit-picking:  You don't need the arguments in super.  I.e.: `super().__init__(fv, fitsimage)` will work.

Reply:
Fixed